### DISCUSSION TITLE
I encountered a problem while writing a smart contract to send ETH. Please provide me with an answer

### DISCUSSION THREAD

**Discussion by Apostleasd** [Cyfrin/foundry-full-course-cu#1727] [Q&A] [answered]
As shown in Figure 1, I have used Solidity to write contracts SendEth for sending ETH and EthReceiver for receiving ETH, and have already sent 1 ETH token to the SendEth contract during deployment
![1](https://github.com/Cyfrin/foundry-full-course-f23/assets/38981397/20a3ccdb-3e35-4a9f-ac0a-03f9c61e9f85)

However, when I called the sendViaTransfer function in SendEth, I found that the address displayed in the log for 'from' was not the address of the SendEth contract, but another address I had not seen before.
I have also considered whether it is possible that it is the caller's address, but if it is the caller's address, why can we use the ETH in the contract address?
![2](https://github.com/Cyfrin/foundry-full-course-f23/assets/38981397/f5954c86-70b3-4b08-8875-a9a63cf1c63f)

**Comment by EngrPips** [comment, marked as answer]
Hello @Apostleasd, The `from` field you saw when you opened the transaction details of the `sendViaTransfer` function does not specify the sender of the ETH but instead, the address that initiated the `sendViaTransfer` transaction. So, the ETH was sent using the `SendEth` contract. The address displayed in the `from` field of the `sendViaTransfer` transaction details is the address your remix was connected to when you clicked on the `sendViaTransfer` function button. 

<img width="371" alt="Screenshot 2024-05-17 at 16 52 38" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/8fd114ba-4c7a-4503-bc81-1fe18d017dd4">

**Reply by Apostleasd** [reply]
OK，I got it，thank you very much!

**Reply by EngrPips** [reply]
I am glad you got it. Keep crushing it, friend.
